src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const resultStates = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === `addProperties`) {
+      Object.assign(newState, actions[i].extraData);
+    }
+
+    if (actions[i].type === `clear`) {
+      for (const key in newState) {
+        delete newState[key];
+      }
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (const key in newState) {
+        if (actions[i].keysToRemove.includes(key)) {
+          delete newState[key];
+        }
+      }
```

Comment:
You can freely skip this check and just loop through all `actions[i].keysToRemove` values. `delete` operator will skip all unexistable keys [without throwing exception](https://stackoverflow.com/a/16406935).